src/driver/include/driver/workload_parsers.hpp

```diff
@@ -16,15 +16,68 @@
 #define HEADER_0369D27D_9A68_4981_B344_4BAB3EE09A80_INCLUDED
 
 #include <boost/filesystem.hpp>
+#include <optional>
 #include <yaml-cpp/yaml.h>
 
 #include <driver/v1/DefaultDriver.hpp>
 
 namespace genny::driver::v1 {
 
-using YamlParameters = std::map<std::string, YAML::Node>;
 namespace fs = boost::filesystem;
 
+/**
+ * Values stored in a Context are tagged with a type to ensure
+ * they aren't used incorrectly.
+ *
+ * If updating, please add to typeNames in workload_parsers.cpp
+ * Enums don't have good string-conversion or introspection :(
+ */
+enum class Type {
+    kParameter,
+    kActorTemplate,
+    kActorInstance,
+};
+
+/**
+ * Manages scoped context for stored values.
+ *
+ * Only create using a ContextGuard.
+ */
+class Context {
+public:
+    using ContextValue = std::pair<YAML::Node, Type>;
+    using Scope = std::map<std::string, std::pair<YAML::Node, Type>>;
+
+    std::optional<YAML::Node> get(const std::string&, const Type& type);
+    void insert(const std::string&, const YAML::Node&, const Type& type);
+
+    // Insert all the values of a node, assuming they are of a type.
+    void insert(const YAML::Node&, const Type& type);
+
+    /**
+     * Opens a scope, closes it when exiting scope.
+     */
+    struct ScopeGuard {
+        ScopeGuard(Context& context) : _context{context} {
+            _context._scopes.emplace_back();
```

Comment:
I had to triple-read this to understand that we're creating a new `Context` here. It's easy to forget that `emplace` forwards to the ctor. Maybe add a comment along the lines of  `// Emplace (push) a new Context onto the stack`. Idk. If I had more C++14 paged into my memory maybe this would have been more obvious.